upgrade base image to debian 12 and tor daemon to 0.4.8.13

m0wer opened the pull request:
Based on https://github.com/lncm/docker-tor/pull/30 but a bit more up to date. Fixes #13 and #21.

I will mantain the fork in my repo until merged. I've enabled CI to automatically build and publish the images to [ghcr.io/m0wer/docker-tor](https://github.com/m0wer/docker-tor/pkgs/container/docker-tor)

```shell
docker pull ghcr.io/m0wer/docker-tor:0.4.8.13
```

The ci commit should be removed before merging unless you also want to publish images to the GitHub image registry.

m0wer committed:
feat: upgrade base image to debian 12 and tor daemon to 0.4.8.13

m0wer committed:
ci: automate Docker image build and push

m0wer committed:
feat: update tor to 0.4.8.14

m0wer committed:
feat: update to 0.4.8.21 with hardcoded SHA256 verification

m0wer committed:
ci: update actions

m0wer committed:
ci: cleanup pipelines

m0wer committed:
feat: enable SocksPort on all interfaces by default

m0wer committed:
feat: Add default torrc configuration file

m0wer committed:
chore: fork repository cleanup

m0wer committed:
feat: Enhance README with control port and hidden service instructions

lnliz commented:
Good to see someone also working on this important repository.
lncm repositories are abandoned but I created and published docker images for tor image based on this repo here [https://github.com/lnliz/docker-tor](https://github.com/lnliz/docker-tor) a bit back.
If you want to add to that it would be great

m0wer commented:
Hi @lnliz ! I went with maintaining the fork and have added some features like a default torrc file that binds to all ports and serves the SOCKS proxy for easy integration into other projects. Also cleaned up a lot of workflows and reworked the readme.

I no longer expect this to be merged and will keep the fork up to date more like an alternative image. The images are now built and published in GitHub.

Thanks for your work as well!

m0wer committed:
feat: Update base image to Debian 13-slim

lnliz commented:
Thank you for your public contributions @m0wer - this is very good and I learned a lot and will use some for my fork too

m0wer committed:
feat: Update Tor to version 0.4.8.22

m0wer committed:
feat: entrypoint script to fix permissions on /var/lib/tor

m0wer committed:
feat: Update Tor to version 0.4.9.5

m0wer committed:
feat: Update Tor to version 0.4.9.6

m0wer committed:
ci: update GitHub Actions versions and add multi-platform build support

- actions/checkout: v5 → v6.0.2
- docker/login-action: v3 → v4.2.0
- docker/metadata-action: v5 → v6.1.0
- docker/build-push-action: v5 → v7.2.0
- Add docker/setup-qemu-action@v4.1.0
- Add docker/setup-buildx-action@v4.1.0
- Build for: linux/amd64, linux/arm64, linux/arm/v7, linux/arm/v6, linux/386, linux/ppc64le, linux/s390x

m0wer committed:
feat: Update Tor to version 0.4.9.8

m0wer committed:
ci: parallelize multi-arch builds and use native arm64 runner

- Build each platform in a matrix job for parallel execution
- Use ubuntu-24.04-arm for native linux/arm64 builds
- Keep QEMU only for non-native architectures
- Add per-arch GHA build cache scopes
- Merge per-arch images into final multi-arch manifests

m0wer committed:
ci: fail release if multi-arch manifest is incomplete

Add post-merge manifest verification for all published tags to ensure
linux/amd64, linux/arm64, linux/arm/v7, linux/arm/v6, linux/386,
linux/ppc64le, and linux/s390x are all present.

m0wer committed:
feat: Update Tor to version 0.4.9.9

m0wer committed:
chore: upgrade tor to 0.4.9.10

m0wer committed:
feat: update Tor to version 0.4.9.11

m0wer committed:
fix: define persistent Tor runtime paths
